### DISCUSSION TITLE
How to set value of a NODE inside the datachange_notification ?

### DISCUSSION THREAD

**Discussion by dibyendumca** [FreeOpcUa/opcua-asyncio#1121] [Show and tell]
**Hello,
I have a requirement as follows.

I want to update the value of a NODE as soon as I get the datachange_notification of another specific NODE.
I have the client code as follows...**

```
import asyncio
from asyncua import ua, Client, Node
from asyncua.client.ua_client import UaClient

class SubHandler(object):
    def __init__(self, clientId):
        self.clientId = clientId

    def datachange_notification(self, node, value, Data):
    if str(node) == "ns=2;i=19":
          curStat = self.clientId.get_node("ns=2;i=6")
          await curStat.set_value(value)
```
**But I could not call this " await curStat.set_value(value) " inside a non async function.
I get the following ERROR...**

await curStat.set_value(value)
    ^
SyntaxError: 'await' outside async function

**Comment by schroeder-** [comment]
You can use a async function aswell:

```Python
class SubHandler(object):
    def __init__(self, clientId):
        self.clientId = clientId

    async def datachange_notification(self, node, value, Data):
    if str(node) == "ns=2;i=19":
          curStat = self.clientId.get_node("ns=2;i=6")
          await curStat.set_value(value)
```

**Reply by dibyendumca** [reply]
Hello @schroeder- 

Thanks very much, it worked.

**Reply by dibyendumca** [reply]
Hello @schroeder- 

How can I get the value of a node within my Server component?
I mean to say, is it possible to write the following code in my Server component?
```
         n = server.get_node("ns=2;i=20")
         val = n.get_value()
```

**Reply by schroeder-** [reply]
the server has the same interface for interacting with the addres space (variables, objects...) as the client.
But I think you you miss the await:
```
n = server.get_node("ns=2;i=20")
val = await n.get_value()
```

**Reply by dibyendumca** [reply]
Hello @schroeder- ,

I have added the following code in my Server Component
`server.get_objects_node().add_variable(ua.NodeId.from_string((ua.NodeId.from_string("ns=2;s=SIMULATION"), "2:SIMULATION", 0)`

I have coded in my Client Component as follows?
```
cNode = client.get_node("ns=2;s=SIMULATION")
handleData = await sub.subscribe_data_change(cNode)
```

But I do not get the datachange_notification for this particular NODE.
Can you please suggest some knowhow?
